samza-core/src/main/java/org/apache/samza/runtime/LocalContainerRunner.java

```diff
@@ -65,13 +76,30 @@ public static void main(String[] args) throws Exception {
     }
     String jobName = jobConfig.getName().get();
     String jobId = jobConfig.getJobId();
-    MDC.put("containerName", "samza-container-" + containerId);
+    MDC.put("containerName", containerName);
     MDC.put("jobName", jobName);
     MDC.put("jobId", jobId);
 
+    writeMetadataFile(jobName, jobId, containerName, System.getenv(ShellCommandConfig.ENV_EXECUTION_ENV_CONTAINER_ID()), config);
+
     ApplicationDescriptorImpl<? extends ApplicationDescriptor> appDesc =
         ApplicationDescriptorUtil.getAppDescriptor(ApplicationUtil.fromConfig(config), config);
 
     ContainerLaunchUtil.run(appDesc, containerId, jobModel);
   }
+
+  public static void writeMetadataFile(String jobName, String jobId, String containerName, String execEnvContainerId, Config config) throws Exception {
```

Comment:
I suppose we need this logic in AM as well to react to OOM in the application master.
Would it be possible to have AM detect that container OOM'd in which case, we have this logic in one place and have the AM react for container/AM OOMing?

Reply:
Its possible, but shouldnt be a common occurrence. 
I can move this to a common place, and have the ClusterBasedJC also call it.

Reply:
done